### PR TITLE
Add Awareness of Aliases To Avoid Duplicate Struct Error Check

### PR DESCRIPTION
**Pull request by rauljordan**
Currently, fastssz does not support the following structure of packages:

```
proto/
  v1/
    block.go
  v2/
    block.go
```

Where both files define a `type BeaconBlock struct` and v2 _includes_ v1. This is important for us in [Prysm](https://github.com/prysmaticlabs/prysm) because we want to use semantic versioning to create new types while leveraging some old code from previous versions to avoid code duplication. Attempting to generate ssz bindings for a package structure above will result in an error found in the `generateIR()` function:

`two structs found with the same name: BeaconBlock`

A workaround for our case was to define aliases such as

```go
package v2

type BeaconBlockV2 struct{}

type BeaconBlock = BeaconBlockV2
```
and call fastssz to generate bindings for BeaconBlockV2 to avoid the duplicate struct error. To achieve this, we needed to make fastssz aware of aliases as well when decoding AST structs.

### CODE REVIEW

**Comment by ferranbt** [comment]
Could you add an example under spectests and regenerate everything with:
```
$ make build-spec-tests
```
This is a safe measure to make sure the new changes do not introduce something unexpected.

**Review by ferranbt — CHANGES_REQUESTED** [review]
Please add an example in the spectests folder and run:
```
$ make build-spec-tests
```
so that we are sure that the change implements the correct behaviour.

**Comment by rauljordan** [comment]
Hi @ferranbt, just pushed the example and it builds as expected

File: spectests/structs.go [hunk 1]
```diff
@@ -197,3 +197,10 @@ type BeaconBlockHeader struct {
 type ErrorResponse struct {
 	Message external.DynamicBytes `ssz-max:"256"`
 }
+
+// BeaconBlockV3 serves to ensure we can have an alias with the same name as an imported package.
+type BeaconBlockV3 struct {
+	innerBlock *external2Alias.BeaconBlockV2
+}
+
+type BeaconBlockV2 = BeaconBlockV3
```
Comment (rauljordan) [inline spectests/structs.go:206]: Previously, this would complain

File: spectests/structs_encoding.go [hunk 2]
```diff
@@ -3650,3 +3650,45 @@ func (e *ErrorResponse) HashTreeRootWith(hh *ssz.Hasher) (err error) {
...
+	if size != 0 {
+		return ssz.ErrSize
+	}
+
+	return err
+}
+
+// SizeSSZ returns the ssz encoded size in bytes for the BeaconBlockV3 object
+func (b *BeaconBlockV3) SizeSSZ() (size int) {
+	size = 0
```
Comment (ferranbt) [inline spectests/structs_encoding.go:3679]: Is this correct?

**Review by ferranbt — CHANGES_REQUESTED** [review]

**Pull request closed without merging**